client/pages/exercises.jsx

```diff
@@ -71,10 +71,11 @@ export default function Exercises(props) {
       },
       body: JSON.stringify(body)
     })
-      .then(response => response.json())
-      .then(data =>
-        clearExercises()
-      )
+      .then(response => {
+        if (!response.ok) window.location.hash = 'error';
+        response.json();
+      })
+      .then(data => clearExercises())
       .catch(err => console.error('ERROR:', err));
     window.location.hash = 'workout';
```

Comment:
There are a few issues here.

- It's not advisable to push the user to a new page when there's an error. instead show the error on this page.
- you are changing the location hash before the fetch finishes